client/ayon_core/pipeline/colorspace.py

```diff
@@ -1637,3 +1649,129 @@ def get_imageio_config(
+
+    """
+    if project_settings is None:
+        project_settings = get_project_settings(project_name)
+
+    imageio_global, imageio_host = _get_imageio_settings(
+        project_settings, host_name
+    )
+
+    builtin_tried = False
+    # Collect custom paths for last step
+    custom_paths = []
+    # Add host path to custom paths
+    host_path = imageio_host.get("ocio_config", {}).get("filepath")
+    if host_path:
+        custom_paths.append(host_path)
+
+    for profile in imageio_global["ocio_config_profiles"]:
+        profile_type = profile["type"]
+        if profile_type == "custom_path":
+            custom_paths.append(profile["custom_path"])
+            continue
+
+        #
+        if profile_type != "builtin_path" or builtin_tried:
+            continue
+
+        template = profile["builtin_path"]
+        try:
+            path = template.format(**os.environ)
+        except Exception:
+            continue
+        if os.path.realpath(path) == os.path.realpath(config_path):
+            return template
+
+    if anatomy is None:
+        anatomy = Anatomy(project_name)
+
+    success, rootless_path = anatomy.find_root_template_from_path(config_path)
+    if success:
+        return rootless_path
+
+    # Try to fill custom paths only with 'os.environ'
+    failed_custom_paths = []
+    for custom_path in custom_paths:
+        try:
+            path = custom_path.format(**os.environ)
+        except Exception:
+            failed_custom_paths.append(custom_path)
+            continue
+
+        if os.path.realpath(path) == os.path.realpath(config_path):
+            return custom_path
+
+    # If none of custom paths formatting failed it means we don't need to try
+    #   with entities template data.
+    if not failed_custom_paths:
+        raise ValueError("Failed to resolve OCIO config template.")
+
+    # Prepare all entities for template data
+    if project_entity is None:
+        project_entity = ayon_api.get_project(project_name)
+    if folder_path and not folder_entity:
+        folder_entity = ayon_api.get_folder_by_path(
+            project_name, folder_path
+        )
+    if task_name and folder_entity and not task_entity:
+        task_entity = ayon_api.get_task_by_name(
+            project_name, folder_entity["id"], task_name
+        )
+    template_data = get_template_data(
+        project_entity,
+        folder_entity,
+        task_entity,
+        host_name,
+        project_settings,
+    )
+    template_data.update(os.environ.items())
+
+    # Try to fill custom paths with template data passed to function
+    for custom_path in failed_custom_paths:
+        try:
+            path = custom_path.format(**template_data)
+        except Exception:
+            continue
+        if os.path.realpath(path) == os.path.realpath(config_path):
+            return custom_path
+
+    raise ValueError("Failed to resolve OCIO config template.")
```

Comment:
Because `get_imageio_config` and `get_imageio_config_preset` does return path based on current settings and current context. This does look into all options that might be used.

Reply:
> Because `get_imageio_config` and `get_imageio_config_preset` does return path based on current settings and current context. This does look into all options that might be used.

Exactly - and those should be the ones we should only consider, right? We shouldn't consider the ones from another profile not related to current context?

Reply:
Then we need the validator... And in that case the logic happening in `create_instances_for_aov` does not happen there, because all you need to to collect config data for the context.